frontend/src/contexts/modalContext.tsx

```diff
@@ -13,42 +13,38 @@ export const ModalContext = React.createContext<IModalContext>({
   setDisplayModal: (display: boolean) => {
     throw Error('setDisplayModal function not defined');
   },
-  setModalProps: (modalProps?: ModalProps) => {
+  setModalContent: (modalProps?: ModalContent) => {
     throw Error('setModalProps function not defined');
   },
 });
 
-export const ModalContextProvider = (
-  props: IModalContext & { children: React.ReactChild | React.ReactChild[] | React.ReactNode },
-) => {
+export const ModalContextProvider = (props: {
+  children: React.ReactChild | React.ReactChild[] | React.ReactNode;
+}) => {
   const [modalProps, setModalProps] = useState<ModalProps | undefined>(undefined);
-  const displayFunction = React.useCallback(
-    (displayModal: boolean) => {
-      setModalProps({ ...modalProps, display: displayModal });
-    },
-    [modalProps],
-  );
+  const [showModal, setShowModal] = useState<boolean>(false);
+
   const updateFunction = React.useCallback(
-    (updatedModalProps?: ModalProps) =>
+    (updatedModalContent?: ModalContent) => {
       setModalProps({
-        ...modalProps,
-        ...updatedModalProps,
-        display: updatedModalProps?.display ?? modalProps?.display ?? false,
-        setDisplay: displayFunction,
-      }),
-    [displayFunction, modalProps],
+        ...updatedModalContent,
```

Comment:
right, ok, so keep the display logic separate from the other modal content since generally the display is the only thing that changes after initialization.

Reply:
that is correct. It also does not make sense to allow the consumer to change/override the display function on the props.